### DISCUSSION TITLE
Is there a way to limit the length of the file being created?

### DISCUSSION THREAD

**Discussion by davidliaos** [absadiki/subsai#29] [Q&A] [answered]
Currently the generation is working out fine but its too long. Its generating subtitles for 0-9 seconds and thats 9 seconds of text, which at the speed that my video is speaking is about two paragraphs worth of text to be showing as subtitles.

Is there a way to limit the text's generation, either by character length, or timeframe? ie generate captions for 2 second intervals, or generate captions for 100 characters only and have it autosync.
Syncing is actually time perfectly but the length is the real issue. 

I've read through the docs but I'm not exactly sure if what I found does what I want it to.
I've found in `config_schema `, `sample_len` and `length_penalty`

Specifically `refine_whisper_precision` in whisper_timestamped_model seems perfect but am unsure how to impliment, and I want to use fast_whisper for speed, but I'm willing to make the switch 

Also is using a mp3 or a mp4 file more efficient? I need to create the mp3 file for another part of the project and in theory its a smaller file which should take less time.

**Comment by davidliaos** [comment]
@abdeladim-s Sorry, not sure if you actively read discussions but the documentation isn't really explaining.

**Comment by absadiki** [comment]
Hi @davidliaos,

I don't quite understand your question. 
* Do you want to generate the transcription for a specific time slot instead of the whole media file ? for example your file is 10 seconds and you want just **from second 2 to second 7** for example ?

* Or is your problem is because your file is large and it is taking so much time to generate the transcription to, you need to do it in a small chunks and then assemble it in one file ?

About your other question, using mp3 or mp4 does not make any difference with regards to the transcription time.

**Comment by davidliaos** [comment]
@abdeladim-s 
I believe the first option is more what I meant, but I want to offer more clarification.

Currently I'm getting these time stamps, 

00:00:00,000 --> 00:00:30,000
(500 characters)
2
00:00:30,000 --> 00:01:00,000
(400 characters)
3
00:01:00,000 --> 00:01:29,960
(500 characters)
4
00:01:29,960 --> 00:01:44,960
(200 characters)

These are fine but they are sending entire paragraphs of text to throw on screen which covers up the actual video itself.
I'm looking to see if I can "limit" the length of the captions so it'll instead maybe have 13 different lines and 10 seconds each instead of 30 seconds each. 

Initially I was looking to see if i could do a character limit, such as each line is limited to 200 characters, and if it reaches 200 characters it'll create a new caption and a new timestamp.

**Reply by absadiki** [reply, marked as answer]
Ahh ok I think I understand, I did this in my [pywhispercpp](https://github.com/abdeladim-s/pywhispercpp) repo. 

To do this in `subsai` you will need to choose the `whisper.cpp` from the models, then from the [configs](https://abdeladim-s.github.io/subsai/#subsai.models.whispercpp_model.WhisperCppModel.config_schema) you'll need to set `token_timestamps` to `True` and `max_len` to the maximum number of characters  you want (in your case 200).

You can try it from the `webui` quickly to see the results.

**Reply by davidliaos** [reply]
Sorry, I'm not really used to delving deep into documentation. 
When using the model and config on the webui it works exactly the way I wanted, its actually is really nicely built too btw.

I'm a bit confused on how to use/impliment the config in a function, and I'm getting a KeyError whenever I try to use whisper.cpp
KeyError still comes up even when its just "model = subs_ai.create_model('whisper.cpp', {'model_type': 'small'})"
Heres my current function and what I assumed from the documentation.
```
def createsrt(path, name):
    file = f'./{path}/{name}.mp4'
    subs_ai = SubsAI()
    model = subs_ai.create_model('whisper.cpp', {'model_type': 'small', 'token_timestamps': True, 'max_len': 200})
    subs = subs_ai.transcribe(file, model)
    srt_folder = './srt'
    srt_path = os.path.join(srt_folder, f'{name}.srt')
    subs.save(srt_path)
   ```

**Reply by absadiki** [reply]
can you try with ? 
```python
model = subs_ai.create_model('ggerganov/whisper.cpp', {'model_type': 'small', 'token_timestamps': True, 'max_len': 200})
```

**Reply by davidliaos** [reply]
Works great, thank you, is there a way to "hide" the whisper progress bar?

**Reply by absadiki** [reply]
I think there is a `verbose` [config](https://abdeladim-s.github.io/subsai/#subsai.models.whisper_model.WhisperModel.config_schema), try to set it to **False**.
